client/my-sites/email/form/mailboxes/components/new-mailbox-list/index.tsx

```diff
@@ -111,25 +98,51 @@ const NewMailBoxList = (
 		submitActionText = translate( 'Submit' ),
 	} = props;
 
-	const existingMailboxes = useGetExistingMailboxNames( provider, selectedDomainName );
+	const existingMailboxes = useGetExistingMailboxNames(
+		provider,
+		selectedDomainName,
+		isInitialMailboxPurchase
+	);
 
-	const createNewMailbox = () =>
-		new MailboxForm< EmailProvider >( provider, selectedDomainName, existingMailboxes );
+	const createNewMailbox = () => {
+		const mailbox = new MailboxForm< EmailProvider >(
+			provider,
+			selectedDomainName,
+			existingMailboxes
+		);
+		// Set initial values
+		Object.entries( initialFieldValues ).forEach( ( [ fieldName, value ] ) => {
+			mailbox.setFieldValue( fieldName as FormFieldNames, value );
+		} );
+		setFieldsVisibilities( [ mailbox ], hiddenFieldNames );
+		return mailbox;
+	};
 
 	const [ mailboxes, setMailboxes ] = useState( [ createNewMailbox() ] );
 	const isTitan = provider === EmailProvider.Titan;
 
-	// Set visibility for desired hidden fields
-	setFieldsVisibilities( mailboxes, hiddenFieldNames );
-	// Set initial values
-	setFieldsInitialValues( mailboxes, initialFieldValues );
+	const persistMailboxesToState = useCallback( () => {
```

Comment:
Do we gain something by wrapping this in `useCallback`?

Reply:
Not a big deal really, but unless we're doing it as a render optimisation for React (which doesn't seem to be the case here), I don't see why it's useful to memoize it.